### DISCUSSION TITLE
pseudo-db, dynamic fields,dynamic types, builtin.object, uuids, metaprogramming, dictionary like

### DISCUSSION THREAD

**Discussion by rd4com** [modular/modular#654] [Show and tell]
First thank to the team to release only 6 days after q/a
here is the repo
https://github.com/rd4com/mojo-relational-objects-exploration
it is also using some structs as wrappers
what do you think ?

**Comment by drunkwcodes** [comment]
I think it is interesting. It is kind of a naive dictionary implementation despite of the name is Db.

I would like to know how far it can go.
Here is my little request: would you like to do the pressure testing for me in mojo?

I have this python script which has performance issue:
https://github.com/drunkwcodes/inp_editing/blob/main/src/inp_editing/inp.py#L32

The `Node` is a data class which has three attributes, xyz.
It is about the coordinates of a point.

I try to create ten thousands of Nodes and give them an unique id.
The script becomes too slow  to be acceptable when I do "if Node in (x, y, z) not exists then create it else get the id" things in 20000 nodes' scale.

So I use sqlite to solve the performance issue. (Please see models.py) 
It only takes 4 minutes to run that script now. you should try it by running inp.py.
But it is sad that dataclass can not be used in this circumstance.

Please try my repo and let's see if mojo has potential to make dataclass more performant to be applicable.

**Comment by rd4com** [comment]
Ok for the test iam working on it,

theres alway improvements that could be done like for searching by uuid:
Each time an item get searched it could be swapped with the left element for faster access on next search.

What would be slower is the deletion of object themselves, it id had iterate the huge list to rebuild a list without it in it.

Field deletion should be plenty fast since we have the object at hand already so it'd depend on how much fields there is because of the same reasons as above.

**Reply by drunkwcodes** [reply]
Good to hear that you have so many thoughts about mojo application. Do you know mojodojo?

**Reply by rd4com** [reply]
i saw the website yes its been usefull to me

**Reply by drunkwcodes** [reply]
The website is actually open source too.

**Comment by drunkwcodes** [comment]
The dataclass implementation runs more than three days, but still not complete.

If mojo's implementation can beat sqlite's performance, it will be a 50000x plus speedup.

It can be a shiny blog post as this [modular blog](https://www.modular.com/blog/mojo-a-journey-to-68-000x-speedup-over-python-part-3)!

Are you sure you can make it?
